specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json

```diff
@@ -0,0 +1,227 @@
+          "parameters": [
+              {
+                  "name": "subscriptionId",
+                  "in": "path",
+                  "required": true,
+                  "type": "string"
+              },
+              {
+                  "name": "resourceGroupName",
+                  "in": "path",
+                  "required": true,
+                  "type": "string"
+              },
+              {
+                  "name": "resourceName",
+                  "in": "path",
+                  "required": true,
+                  "type": "string"
+              }
+          ],
+          "responses": { "200" : { "description": "Success" } } 
+        },
+        "put": {
+          "tags": [ "NotebookResource" ],
+          "operationId": "CreateOrUpdateNotebookResource",
+          "consumes": [ "application/json-patch+json", "application/json", "text/json", "application/*+json" ],
+          "produces": [],
+          "parameters": [
+            {
+              "name": "subscriptionId",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "resourceGroupName",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "resourceName",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "resourceRequest",
+              "in": "body",
+              "required": false,
+              "schema": { "$ref": "#/definitions/NotebookResource" }
+            }
+          ],
+          "responses": { 
+            "200": { 
+              "description": "Success" 
+            }, 
+            "201": {
+              "description": "Success"
+            } 
+          }
+        },
+        "delete": {
+          "tags": [ "NotebookResource" ],
+          "operationId": "DeleteNotebookResource",
+          "consumes": [],
+          "produces": [],
+          "parameters": [
+            {
+              "name": "subscriptionId",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "resourceGroupName",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            },
+            {
+              "name": "resourceName",
+              "in": "path",
+              "required": true,
+              "type": "string"
+            }
+          ],
+          "responses": { 
+            "200": { 
+              "description": "Success" 
+            }, 
+            "204": {
+              "description": "Success"
+            } 
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Notebooks/NotebookProxies/{resourceName}/updatePrivateLinkId/{privateLinkId}": {
```

Comment:
what is this API supposed to do?

Reply:
This API is updating the resource by adding a private link Id.